src/hotspot/os/linux/gc/z/zPhysicalMemoryBacking_linux.cpp

```diff
@@ -204,12 +207,13 @@ int ZPhysicalMemoryBacking::create_mem_fd(const char* name) const {
   snprintf(filename, sizeof(filename), "%s%s", name, ZLargePages::is_explicit() ? ".hugetlb" : "");
 
   // Create file
-  const int extra_flags = ZLargePages::is_explicit() ? MFD_HUGETLB : 0;
+  const int extra_flags = ZLargePages::is_explicit() ? (MFD_HUGETLB | MFD_HUGE_2MB) : 0;
   const int fd = ZSyscall::memfd_create(filename, MFD_CLOEXEC | extra_flags);
   if (fd == -1) {
     ZErrno err;
     log_debug_p(gc, init)("Failed to create memfd file (%s)",
-                          ((ZLargePages::is_explicit() && err == EINVAL) ? "Hugepages not supported" : err.to_string()));
+                          (ZLargePages::is_explicit() && (err == EINVAL || err == ENODEV)) ?
+                          "Hugepages (2M) not supported" : err.to_string());
```

Comment:
Maybe this should be something like:

```suggestion
                          "Hugepages (2M) not available" : err.to_string());
```

As in `ZArguments` the code already checks that 2M page size are requested (if any).

Reply:
Fixed